graspologic/plot/plot.py

```diff
@@ -673,6 +677,253 @@ def pairplot(
+    corresponds to x-axis
+    and k corresponds to the y-axis onto the axis that is ax. plot_ellipse then
+    applies a gmm ellipse onto the scatterplot
+    using the data from Y_, means, covariances.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        Input data.
+    Y_ : array-like, shape (n_samples, 1)
+        When no labels are passed into pairplot_with_gmm then these are the gmm
+        lables.
+        When labels are passed into pairplot_with_gmm then these are numbers
+        that have corrsponding
+        relationship to the labels
+    means : array-like, shape (n_components, n_features)
+        Estimated means from gmm
+    covariances : array-like, shape (with 'full') (n_components, n_features, n_features)
+        estimated covariances from gmm
+    ax : axis object
+        The location where plot_ellipse will plot
+    j : int
+        column index of feature of interest from X which will be the x-axis data.
+    k : int
+        column index of feature of interest from X which will be the y-axis data.
+    colors : array-like or list,seaborn color palette
+    lab_dict : dict
+        Dictionary of labels mapped to unique number
+    ind_to_lab : dict
+        The reverse of lab_dict
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    palette : str, dict, optional, default: 'Set1'
+    """
+    for i, (mean, covar, color) in enumerate(zip(means, covariances, colors)):
+        v, w = linalg.eigh(covar)
+        v = 2.0 * np.sqrt(2.0) * np.sqrt(v)
+        u = w[0] / linalg.norm(w[0])
+        if lab_dict and ind_to_lab:
+            # Plot the scatter
+            sns.scatterplot(
+                X[Y_ == i, j],
+                X[Y_ == i, k],
+                ax=ax,
+                label=ind_to_lab[i],
+                color=colors[i],
+            )
+            # Plot an ellipse to show the Gaussian component
+            angle = np.arctan(u[1] / u[0])
+            angle = 180.0 * angle / np.pi
+            ell = mpl.patches.Ellipse(
+                [mean[j], mean[k]], v[0], v[1], 180.0 + angle, color=colors[i]
+            )
+        else:
+            if not np.any(Y_ == i):
+                continue
+            sns.scatterplot(
+                X[Y_ == i, j],
+                X[Y_ == i, k],
+                ax=ax,
+                label="Block {}".format(i),
+                color=colors[i],
+            )
+            # Plot an ellipse to show the Gaussian component
+            angle = np.arctan(u[1] / u[0])
+            angle = 180.0 * angle / np.pi
+            ell = mpl.patches.Ellipse(
+                [mean[j], mean[k]], v[0], v[1], 180.0 + angle, color=colors[i]
+            )
+        ell.set_clip_box(ax.bbox)
+        ell.set_alpha(alpha)
+        ax.add_artist(ell)
+        ax.set_xticks([]), ax.set_yticks([])
+        ax.set_ylabel(k), ax.set_xlabel(k)
+
+
+def pairplot_with_gmm(
```

Comment:
This was discussed in class today and it was said that once this PR meets the necessary standard it will be merged and then i will work on trying to use pairplot from graspologic to achieve this same thing.